swap_meet/vendor.py

```diff
@@ -0,0 +1,52 @@
+class Vendor:
+    def __init__(self, inventory=None):
+        self.inventory = inventory
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.item = item
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        self.item = item
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        items_matching_category = []
+        for item in self.inventory:
+            if category == item.category:
+                items_matching_category.append(item)
+        return items_matching_category
+
+    def swap_items(self, Vendor, my_item, their_item):
+        if (my_item in self.inventory) and (their_item in Vendor.inventory):
+            self.remove(my_item)
+            self.add(their_item)
+            Vendor.add(my_item)
+            Vendor.remove(their_item)
+            return True
+        return False
+
+    def swap_first_item(self, Vendor):
+        if (self.inventory and Vendor.inventory):
+            return self.swap_items(Vendor, self.inventory[0], Vendor.inventory[0])
+        return False
```

Comment:
Hm... this passed the test however I'd heavily caution in using a Class to add and remove items from. "self" refers to an **instance** of the Vendor class so when you're adding and removing items from a data structure in the **Class** itself, you're also affecting the instances. `Vendor` should instead be referencing another **instance**. Book a 1:1 with me anytime you'd like to talk about this (: